clarity/fuzz/fuzz_targets/fuzz_sanitize.rs

```diff
@@ -0,0 +1,342 @@
+        if version >= 32 {
+            return Err(arbitrary::Error::IncorrectFormat);
+        }
+        let data = Arbitrary::arbitrary(u)?;
+        Ok(FuzzStandardPrincipal(StandardPrincipalData(version, data)))
+    }
+}
+
+impl arbitrary::Arbitrary<'_> for FuzzClarityValue {
+    fn arbitrary(u: &mut arbitrary::Unstructured<'_>) -> arbitrary::Result<Self> {
+        let clar_type = u8::arbitrary(u)?;
+        let clar_value = match clar_type {
+            0 => ClarityValue::Int(i128::arbitrary(u)?),
+            1 => ClarityValue::UInt(u128::arbitrary(u)?),
+            2 => ClarityValue::Bool(bool::arbitrary(u)?),
+            3 => ClarityValue::some(FuzzClarityValue::arbitrary(u)?.0)
+                .map_err(|_| arbitrary::Error::IncorrectFormat)?,
+            4 => ClarityValue::none(),
+            5 => ClarityValue::okay(FuzzClarityValue::arbitrary(u)?.0)
+                .map_err(|_| arbitrary::Error::IncorrectFormat)?,
+            6 => ClarityValue::error(FuzzClarityValue::arbitrary(u)?.0)
+                .map_err(|_| arbitrary::Error::IncorrectFormat)?,
+            7 => ClarityValue::Principal(PrincipalData::Standard(FuzzStandardPrincipal::arbitrary(u)?.0)),
+            8 => ClarityValue::Principal(PrincipalData::Contract(QualifiedContractIdentifier::new(
+                FuzzStandardPrincipal::arbitrary(u)?.0,
+                FuzzContractName::arbitrary(u)?.0,
+            ))),
+            // utf8
+            9 => ClarityValue::string_utf8_from_bytes(Arbitrary::arbitrary(u)?)
+                .map_err(|_| arbitrary::Error::IncorrectFormat)?,
+            // ascii
+            10 => ClarityValue::string_ascii_from_bytes(Arbitrary::arbitrary(u)?)
+                .map_err(|_| arbitrary::Error::IncorrectFormat)?,
+            // buff
+            11 => ClarityValue::buff_from(Arbitrary::arbitrary(u)?)
+                .map_err(|_| arbitrary::Error::IncorrectFormat)?,
+            // list
+            12 => {
+                let value_vec: Vec<FuzzClarityValue> = Arbitrary::arbitrary(u)?;
+                ClarityValue::cons_list_unsanitized(value_vec.into_iter().map(|x| x.0).collect())
+                    .map_err(|_| arbitrary::Error::IncorrectFormat)?
+            },
+            // tuple
+            13 => {
+                let tuple_data: Vec<(FuzzClarityName, FuzzClarityValue)> = Arbitrary::arbitrary(u)?;
+                TupleData::from_data(
+                    tuple_data
+                        .into_iter()
+                        .map(|(key, value)| (key.0, value.0))
+                        .collect()
+                )                    
+                    .map_err(|_| arbitrary::Error::IncorrectFormat)?
+                    .into()
+            },
+            _ => return Err(arbitrary::Error::IncorrectFormat),
+        };
+
+        Ok(FuzzClarityValue(clar_value))
+    }
+}
+
+pub fn strict_admits(me: &TypeSignature, x: &ClarityValue) -> Result<bool, CheckErrors> {
```

Comment:
I notice there's no recursion depth checking here.  Could this lead to a stack overflow, given that both `me` and `x` are generated from the fuzzer?

Reply:
Like below, it shouldn't. `TypeSignature` construction (which uses the `TypeSignature::type_of()` method) will not construct a deep enough type signature. The tuple bug fixed in 2.4 also cannot trigger a stack overflow here -- the length is compared before descent.

If it did result in a stack overflow, the fuzzer would (correctly) raise the input case as a failure case.